File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;
class Solution
{
    public int solution(String s)
    {
        Stack stack = new Stack();
        stack.push(s.charAt(0));
        for(int i=1;i<s.length();i++) {
            char c = s.charAt(i);
            if(!stack.isEmpty() && stack.peek()==c){
                stack.pop();
                continue;
            }
            else{
                stack.push(c);
            }
        }
        if(stack.isEmpty())
            return 1;
        return 0;
    }
}
class Stack 
{
    private LinkedList<Integer> stack;
    
    public Stack() {
        stack = new LinkedList<>();
    }
    
    public void push(int n) {
        stack.addLast(n);
    }
    public int pop() {
        return stack.removeLast();
    }
    public int peek() {
        return stack.getLast();
    }
    public boolean isEmpty() {
        return stack.isEmpty();
    }
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s)
{
    var answer = 0;
    let arr = s.split('')
    let stack = []
    for(let i=0;i<arr.length;i++) {
        if(stack.length >0 && (arr[i]===stack[stack.length-1])) {
            stack.pop()
        } else{
            stack.push(arr[i])
        }
    }
    if(stack.length===0) return 1
    return answer;
}